### PR TITLE
python2Packages.faulthandler: 3.1 -> 3.2 and fix build; python2Packages.pytest-faulthandler: remove

### PR DESCRIPTION
**Pull request by bhipple**
It is currently failing on master and 20.03. I spent some time reading the src
code but was not able to figure out why the env var activation is not working.
Since this is currently failing, and since it's dying alongside python 2 anyways
I propose we just disable the 1 failing test.

There's some more information inline in the comment if someone wants to fix this
or dig further.

ZHF: #80379

CC @FRidh @singron I tried the obvious stuff, like unsetting the windows argument or the pip flags, but it didn't seem to make a difference.  Perhaps this makes more sense to you?

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bhipple** [comment]
Also removing `pytest-faulthandler`, since per upstream it should no longer be used w/ pytest >= 5.0:
https://github.com/pytest-dev/pytest-faulthandler

**Comment by bhipple** [comment]
```
$ nix-env -f /home/bhipple/.cache/nixpkgs-review/rev-3f921f3a459c6548c70278c6d53ada0e92466dfa/nixpkgs -qaP --xml --out-path --show-trace --meta
44 package updated:
bcc bpftrace can-isotp chipsec digimend-unstable digitalbitbox evdi-unstable evdi-unstable gdata-sharp glibc libselinux mba6x_bl nix-linter-unstable nixpart nixpart php-couchbase php-couchbase php-pcs php-pcs php-php_excel php-php_excel php-php_excel php-php_excel php-php_excel php-php_excel php-v8 php-v8 php-v8 php-v8 php-v8 php-v8 php-v8js php-v8js prl-tools pyblock pyblock python-notify python-notify python2.7-faulthandler (3.1 → 3.2) python2.7-habanero python2.7-handout python2.7-hass-nabucasa qes r8125

3 package removed:
python2.7-pytest-faulthandler (†9.003.02) python3.7-pytest-faulthandler (†9.003.02) python3.8-pytest-faulthandler (†9.003.02)

$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/bhipple/.cache/nixpkgs-review/rev-3f921f3a459c6548c70278c6d53ada0e92466dfa/build.nix
[1 built, 0.0 MiB DL]
43 package marked as broken and skipped:
digitalbitbox gdata-sharp glibcCross linuxPackages_4_4.evdi linuxPackages_hardkernel_4_14.bcc linuxPackages_hardkernel_4_14.bpftrace linuxPackages_hardkernel_4_14.can-isotp linuxPackages_hardkernel_4_14.chipsec linuxPackages_hardkernel_4_14.digimend linuxPackages_hardkernel_4_14.evdi linuxPackages_hardkernel_4_14.mba6x_bl linuxPackages_hardkernel_4_14.prl-tools linuxPackages_hardkernel_4_14.r8125 nix-linter php72Packages-unit.php_excel php72Packages-unit.v8 php72Packages-unit.v8js php72Packages.php_excel php72Packages.v8 php72Packages.v8js php73Packages-unit.php_excel php73Packages-unit.v8 php73Packages.php_excel php73Packages.v8 php74Packages-unit.couchbase php74Packages-unit.pcs php74Packages-unit.php_excel php74Packages-unit.v8 php74Packages.couchbase php74Packages.pcs php74Packages.php_excel php74Packages.v8 python27Packages.habanero python27Packages.handout python27Packages.hass-nabucasa python37Packages.nixpart python37Packages.notify python37Packages.pyblock python38Packages.libselinux python38Packages.nixpart python38Packages.notify python38Packages.pyblock qes

1 package built:
python27Packages.faulthandler
```
OK now I think we have something reasonable to review :)

File: pkgs/development/python-modules/faulthandler/disable-env-test.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+index e5a2811..16e7a0f 100644
+--- a/tests.py
++++ b/tests.py
+@@ -394,7 +394,7 @@ class FaultHandlerTests(unittest.TestCase):
+             print(faulthandler.is_enabled())
+             """,
+             env=dict(os.environ, PYTHONFAULTHANDLER='x'))
+-        self.assertEqual(output, ["True"])
++        #self.assertEqual(output, ["True"])
+         self.assertEqual(exitcode, 0)
```
Comment (jonringer) [inline pkgs/development/python-modules/faulthandler/disable-env-test.patch]: pytest should be able to run unittest test suites, and it allows for you to easily filter a few tests, but pathcing also works.

**Review by jonringer — APPROVED** [review]
LGTM

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/83466
1 package marked as broken and skipped:
pants

1 package built:
python27Packages.faulthandler
```

**Pull request merged**